iModelCore/ecobjects/src/ECInstance.cpp

```diff
@@ -2687,6 +2698,12 @@ struct  InstanceXmlReader
                 return InstanceReadStatus::ECSchemaNotFound;
                 }
 
+            if (ECObjectsStatus::SchemaIsPruned == schemaStatus)

```

Comment:
This check should be before the check for null schema ... I know the current behavior is for GetSchema to return the fallback schema but I think I'd even check for schema pruned and return directly when calling GetSchema.  

I assume the null check for the schema is old code because the current behavior makes that impossible.  If I were to clean things up more I would probably rewrite this code so that GetSchema does not return the fallback schema and we explicitly use it when necessary.  I don't think we should do that right now because we should just wrap this up ... but moving the pruned check to the call to GetSchema will make the code more resilient.

Reply:
I moved this check to be before the null schema check.